### ISSUE TITLE
The rules for conformity for emotion recognition. 

### ISSUE THREAD

**Issue by chercheurkg** [s3prl/s3prl#528]
In the official version of SUPERB paper; available on its website (https://superbbenchmark.org/), under the subheading - 2.4. Paralinguistics - the line number 7 reads that the evaluation metric for emotion classification is **_accuracy_**. However, the introductory video on the same website says the evaluation metric for emotion classification  is **_unweighted accuracy_**;  after the video runs for 11 minutes and 32 seconds. Which one is the correct evaluation metric for emotion classification? 

The same video also says that both improvised and non improvised versions of the IEMOCAP database were used for the SUPERB benchmark. Which experiment's result to be reported: experiment conducted on improvised or non-improvised version of IEMOCAP dataset? 

This file  - https://github.com/s3prl/s3prl/blob/main/s3prl/downstream/emotion/config.yaml - reads 30000 training steps for emotion recognition. Does it mean 30000 training steps per fold of IEMOCAP dataset? 

Is the best value of evaluation metric from each fold or the average value of evaluation metric from each fold to be selected? How is the final result after the five folds are completed calculated? 

Is there any consolidated list of rules for emotion recognition to conform with SUPERB benchmark? 

should the 'excited' and 'happy' classes of IEMOCAP dataset merged together to comply with the SUPERB benchmark?

**Comment by leo19941227** [comment]
Hi,

1. We simply use accuracy. That was a mistake in the video.
2. We do not conduct separate experiments for improvised and scripted. Both types exist in the train/valid/test sets.
3. Yes
4. For each fold, we train on the train set, evaluate on dev set, across several learning rates, and find the best dev result on this specific fold, and then report the test result on this fold. This process is repeated for 5 times. Finally, 5 testing acc are avearged.
5.  I am not very sure what do you mean by consolidated list of rules. I think the best practice would be following our recipe as described [here](https://github.com/s3prl/s3prl/blob/main/s3prl/downstream/docs/superb.md#er-emotion-recognition)
6. Yes.

**Comment by chercheurkg** [comment]
@leo19941227 ,
Thanks so very much for your reply!! 

1. Which ratio is the dataset split into  train/valid/test sets -  train( 72% ), valid (8%) , test ( 20%) of total number of data (audio files) ?

2. Could you please share the link to the script written for IEMOCAP data split (train/valid/test sets.) or load?

3. It says 30000 steps are to be used for each fold. Does that mean approximately 215 epoch per fold or can you please tell me the exact number of epoch per fold ? 

4. Is only the learning rate allowed to be tuned ? or can other hyper-parameters like batch size be tuned as well ? 

5. is it correct that the parameters for encoders of pre-trained LLM are not allowed to be fine-tuned or trained with the IEMOCAP data for downstream tasks ?

**Comment by leo19941227** [comment]
Hi,

Sorry for the late reply, I did not notice this follow up question.

1 and 2:

There are 5 sessions in IEMOCAP, we first select one session as the testing set, and use the remaining 4 sessions as training and validation set. The data split for train/valid utilizes the random split function (which might not be a very rigorous decision when looking back now) as show here:

https://github.com/s3prl/s3prl/blob/4517a5db118b0e366190ff31a3af9186de255e42/s3prl/downstream/emotion/expert.py#L45

The train/valid ratio is 0.8/0.2 on the remaining 4 sessions utterances.

3. I am not very sure about the epoch number. W decided the total training steps by making sure that the development acc is not going up any further but not setting a fixed number of epoch. However, you can still find out the epoch number by tracing the code.

4. I think only the learning rate is allowed to be tuned.

5. In SUPERB's protocol, yes.

**Comment by chercheurkg** [comment]
@leo19941227,
Thanks for your reply! 
Is it allowed to apply dropouts to linear classifier head?

**Comment by leo19941227** [comment]
Hi,

Well, in principle, since in SUPERB our goal is comparing the quality of representation, so that ideally all the remaining training pipeline other than the representation should be fixed. Or else we are not sure whether the improvement comes from the representation or the newly introduced dropout.
